src/platform/activation_abi.cpp

```diff
@@ -0,0 +1,69 @@
+#include "pal_internal.h"
+#include "opaque_string_wrapper.h"
+#include "platform_activation.h"
+
+namespace xlang::impl
+{
+    template <typename char_type>
+    void* get_activation_factory(
+        xlang_string class_name,
+        xlang_guid const& iid)
+    {
+        for (auto current_namespace = enclosing_namespace(to_string_view<char_type>(class_name));
+            !current_namespace.empty();
+            current_namespace = enclosing_namespace(current_namespace))
+        {
+            xlang_pfn_lib_get_activation_factory pfn = try_get_activation_func(current_namespace);
+            if (pfn)
+            {
+                void* factory{};
+                xlang_result result = (*pfn)(class_name, iid, &factory);
+                if (result == xlang_error_ok)
```

Comment:
fyi - the PAL doc has a some mixed case error definitions (e.g., Xlang_INVALID_ARG)